rust/processor/src/models/fungible_asset_models/v2_fungible_asset_balances.rs

```diff
@@ -62,6 +66,86 @@ pub struct CurrentFungibleAssetBalance {
     pub token_standard: String,
 }
 
+#[derive(Clone, Debug, Deserialize, FieldCount, Identifiable, Insertable, Serialize, Default)]
+#[diesel(primary_key(storage_id))]
+#[diesel(table_name = current_unified_fungible_asset_balances)]
+#[diesel(treat_none_as_null = true)]
+pub struct CurrentUnifiedFungibleAssetBalance {
+    pub storage_id: String,
+    pub owner_address: String,
+    // metadata address for (paired) Fungible Asset
+    pub asset_type: String,
+    pub coin_type: Option<String>,
+    pub is_primary: Option<bool>,
+    pub is_frozen: bool,
+    pub amount_v1: Option<BigDecimal>,
+    pub amount_v2: Option<BigDecimal>,
+    pub last_transaction_version_v1: Option<i64>,
+    pub last_transaction_version_v2: Option<i64>,
+    pub last_transaction_timestamp_v1: Option<chrono::NaiveDateTime>,
+    pub last_transaction_timestamp_v2: Option<chrono::NaiveDateTime>,
+}
+
+fn get_paired_metadata_address(coin_type_name: &str) -> String {
+    if coin_type_name == APTOS_COIN_TYPE_STR {
+        APT_METADATA_ADDRESS_HEX.clone()
+    } else {
+        let mut preimage = APT_METADATA_ADDRESS_RAW.to_vec();
+        preimage.extend(coin_type_name.as_bytes());
+        preimage.push(0xFE);
+        format!("0x{}", hex::encode(sha3_256(&preimage)))
+    }
+}
+
+fn get_primary_fungible_store_address(
```

Comment:
Can you add unit tests for these 2 functions

Reply:
kk

Reply:
is_primary test covers the 2nd.